### DISCUSSION TITLE
Propagate signals between different Seq servers

### DISCUSSION THREAD

**Discussion by Tthongsy** [datalust/seq-tickets#1267] [Help] [answered]
Is there a way to propagate signals between different Seq servers on version 4.2. I couldn't find a way to share the signal

Thanks,
Tommy

**Comment by nblumhardt** [comment, marked as answer]
Hi! On 4.2, probably not, unfortunately.

For recent versions, `seqcli signal list --json` and `seqcli signal import` will do this.

**Reply by Tthongsy** [reply]
Hi,
I think I'll just manually create the signals for now then. 

Thanks,
Tommy

**Reply by nblumhardt** [reply]
Thanks for the heads-up!

There are MASSIVE :-) improvements between Seq 4.2 and the latest versions; if you need any help upgrading, please don't hesitate to drop us a line, we're happy walking you through it or helping with the process wherever possible.

**Reply by Tthongsy** [reply]
Hey Nick,

Just to clarify, **seqcli signal list --json** is used to create the json list with signals and **seqcli signal import** is used to import signals onto the instance of seq that I need?

When I perform those commands on cli, I get these errors:
**The command failed: Error 12029 calling WINHTTP_CALLBACK_STATUS_REQUEST_ERROR, 'A connection with the server could not be established'**

**The command failed: No connection could be made because the target machine actively refused it.**

I'm unsure of what the issue?

Thanks,
Tommy T

**Reply by nblumhardt** [reply]
That's correct; you'll need to specify `-s https://your-server` for each of the commands to point them at the right Seq instance. It looks like the command is trying to access the Seq API at the wrong location - does this help?

It's not certain that they'll work properly with Seq 4.2, however.

**Reply by Tthongsy** [reply]
Oh I see, I didn't know that I had to specify the server, that helped a lot! I have another question, I can't see on the documentation how to log in, as it keeps requesting me to log in. How would I log in through cli , would I be using an API key or logging in by using my user/pass that I use for the url?

Thanks,
Tommy T

**Reply by Tthongsy** [reply]
Sorry to double reply, but I wanted to ask if I also did not have a key setup, do I leave the argument blank or do I need to set one up?

**Reply by nblumhardt** [reply]
Great, glad that helped :+1:

Using an API key is the way to go; you'll need to make sure the key has at least `Read` and `Write` permissions.

If your server doesn't require authentication, you don't need to specify the argument.

**Reply by Tthongsy** [reply]
Hey Nick,

So I tried to do the command, **seqcli signal list --json -s https://your-server**, however it keeps requesting me to log in although there is no API key needed. I'm unsure what I'm doing wrong here.

Thanks,
Tommy T

**Reply by nblumhardt** [reply]
Hi Tommy,

By "no API key needed", do you mean there's no authentication set up on the instance at all?

The "Require an API key..." setting on the Settings >API keys screen only applies to ingestion of log events, and doesn't determine whether or not other functions will require authentication.

It sounds like your server has authentication enabled, in which case you'll need to specify an API key.

Let me know if this helps, or if I've misunderstood where the problem lies!

Best regards,
Nick

**Reply by Tthongsy** [reply]
Hey Nick,

I might be confused myself, is it an API key that was made when the instances are created? As far as I know from my end, the instances actually weren't setup with any sort of authentication so I'm unsure if I may be confused on what type of authentication I need.

Thanks,
Tommy T

**Reply by nblumhardt** [reply]
Hmm, doesn't sound quite right; I wonder if it could be some incompatibility between `seqcli` and the (assuming 4.2?) version of Seq you're using....

You can check whether authentication is enabled by navigating to _Settings > Users_ - if auth isn't enabled, there will be a notice on that page.

Does it work if you create an API key? Should only be a few clicks, under _Settings > API keys_, though you'll need to take care that the correct permissions are selected.

Let me know if this sparks any ideas. Sorry to not have more concrete advice; if the above don't work, we might need to set up a call of some kind to work through it.

**Reply by Tthongsy** [reply]
I think you might be correct that there is auth enabled because on users page there isn't a notice saying that auth isn't enabled.

I guess to clarify, if I create an API key would I be able to remove it afterwards?

**Reply by Tthongsy** [reply]
Sorry for double post

I just got it to work.
Is there a way to save the result from the **seqcli signal list --json -s https://your-server** command as a text file?

**Reply by nblumhardt** [reply]
Great! 👍 

Using `>` redirection in your shell should do the trick:

```
seqcli signal list --json -s https://your-server > signals.txt
```

**Reply by Tthongsy** [reply]
Hey,
The signals end up duplicating when importing signals. how do I use the merge command?

seqcli signal import -s https://myserver -a APIKEY -i test.json
In this case, where would I end up placing the merge command with the release of 2020.3?

**Reply by nblumhardt** [reply]
Hi - the following should do it:

```
seqcli signal import -s https://myserver -a APIKEY -i test.json --merge
```

**Reply by Tthongsy** [reply]
Everything is working!

Thanks for everything! :)

**Reply by nblumhardt** [reply]
🎉 You're welcome!
